R/expr-node.R

```diff
@@ -203,10 +203,14 @@ as_pairlist <- function(x) {
 #'   node of pairlists and an empty pairlist is thus the `NULL`
 #'   object itself.
 #'
-#' * `is_node()` checks that `x` has type `pairlist`.
+#' * `is_pairlist_node()` checks that `x` has type `pairlist`.
```

Comment:
Is that better than just `is_pairlist()`?

Reply:
`is_pairlist()` is taken since rlang 0.1 for the the pairlist data structure: either a pairlist node or `NULL`. Maybe it should be the other way around? `is_pairlist()` to check for pairlist base type? But then I don't know how to call the predicate that checks for either pairlist or `NULL`.

I don't remember having used `is_pairlist()` for checking NULLs as well, so we could just remove that meaning for now until the need for it comes up.

Reply:
I'm confused. What's the difference between `is_pairlist()` and `is_pairlist_node()`, just whether or not `NULL` is included?

Reply:
yes `is_pairlist()` also returns true for NULL while `is_pairlist_node()` only returns true for cons cells.

Reply:
How about:

- `is_pairlist()` for pairlist type only
- `is_node_list()` for pairlist + NULL types
- `is_node()` for pairlist + language types

Reply:
Sounds good